google-cloud-secret_manager/samples/acceptance/create_regional_secret_with_tags_test.rb

```diff
@@ -0,0 +1,124 @@
+# Copyright 2025 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "uri"
+
+require_relative "regional_helper"
+
+describe "#create_regional_secret_with_tags", :regional_secret_manager_snippet do
+  let(:tag_key_short_name) { "my-tag-key-#{(Time.now.to_f * 1000).to_i}" }
+  let(:tag_value_short_name) { "my-tag-value-#{(Time.now.to_f * 1000).to_i}" }
+  let(:tag_parent) { "projects/#{project_id}" }
+  let(:tag_key_client) { Google::Cloud::ResourceManager::V3::TagKeys::Client.new }
+  let(:tag_value_client) { Google::Cloud::ResourceManager::V3::TagValues::Client.new }
+
+  let :tag_key do
+    tag_key_client.create_tag_key(
+      tag_key: {
+        parent: tag_parent,
+        short_name: tag_key_short_name
+      }
+    ).wait_until_done!
+    tag_key_client.list_tag_keys(parent: tag_parent).to_a.find { |key| key.short_name == tag_key_short_name }
+  end
+
+  let :tag_value do
+    tag_value_client.create_tag_value(
+      tag_value: {
+        parent: tag_key.name,
+        short_name: tag_value_short_name
+      }
+    ).wait_until_done!
+    tag_value_client.list_tag_values(parent: tag_key.name).to_a.find { |value| value.short_name == tag_value_short_name }
+  end
+
+  let(:interval_duration) { 15 }
+  let(:max_duration) { 60 }
+
+  def cleanup_tag_value
+    return unless defined?(tag_value) && tag_value
```

Comment:
`defined?(tag_value)` should always be true (as a method) since you've defined it above.

Same for the other test below.